cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -222,6 +254,23 @@ public void resume() throws Exception {
   protected void shutDown() throws Exception {
     httpService.stopAndWait();
     deleteLocalDatasets();
+    destroyWorkflow();
+  }
+
+  private void destroyWorkflow() {
+    TransactionContext transactionContext = basicWorkflowContext.getDatasetCache().newTransactionContext();
+    try {
+      transactionContext.start();
+      workflow.destroy();
+      transactionContext.finish();
+    } catch (Throwable t) {
+      LOG.error("Failed to execute the destroy method of the Workflow.", t);
```

Comment:
similar comment, would be nice to have the workflow name and namespace in the error message